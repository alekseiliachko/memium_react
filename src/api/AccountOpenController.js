import APIController from "./ApiController";

class АccountOpenController extends APIController {
  constructor() {
    super();
  }

  async getAccountsDetails(accountId) {
    const details = await this.request(
      "get",
      `/open/account/details/${accountId}`
    );
    return details;
  }

  async getAccountsAvatar(accountId) {
    const avatar = await this.request(
      "get",
      `/open/account/avatar/${accountId}`,
      {
        responseType: "blob",
      }
    );
    return URL.createObjectURL(avatar.data);
  }
}

export default new АccountOpenController();
