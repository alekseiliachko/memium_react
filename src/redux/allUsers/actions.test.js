import {
  setUserAvatar,
  setUserData,
  fetchUserAvatar,
  fetchUserData,
} from "./actions";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
jest.mock("../../api/AccountOpenController");
import AccountOpenController from "../../api/AccountOpenController";

const mockStore = configureMockStore([thunk]);

const userData = {
  accountId: "22bb3fe5-185c-4c87-8633-1586586506b7",
  name: null,
  bio: "populated",
  gender: "gender",
  dob: 1615046987793,
  avatar: "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
};

describe("actions", () => {
  let store;
  beforeEach(() => {
    store = mockStore({ userDataById: {} });
  });
  it("should create set_user_data action", () => {
    const userId = "32df86b7-872d-45d5-8a51-e217f0bc973d";
    store.dispatch(setUserData(userId, userData));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("should create set_avatar_data action", () => {
    const avatar =
      "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517";
    const userId = "22bb3fe5-185c-4c87-8633-1586586506b7";
    store.dispatch(setUserAvatar(userId, avatar));
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe("async actions", () => {
  let store;
  beforeEach(() => {
    store = mockStore({ userDataById: {} });
  });

  it("create SET_USER_DATA when fetching has been done", () => {
    const accountId = "22bb3fe5-185c-4c87-8633-1586586506b7";
    AccountOpenController.getAccountsDetails.mockResolvedValue({
      data: { userData },
    });
    return store.dispatch(fetchUserData(accountId)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it("create SET_AVATAR_DATA when fetching has been done", () => {
    const accountId = "22bb3fe5-185c-4c87-8633-1586586506b7";
    AccountOpenController.getAccountsAvatar.mockResolvedValue({
      avatar: "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
    });
    return store.dispatch(fetchUserAvatar(accountId)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
