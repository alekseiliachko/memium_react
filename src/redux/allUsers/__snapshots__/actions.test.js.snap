// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create set_avatar_data action 1`] = `
Array [
  Object {
    "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
    "type": "SET_USER_AVATAR",
    "userId": "22bb3fe5-185c-4c87-8633-1586586506b7",
  },
]
`;

exports[`actions should create set_user_data action 1`] = `
Array [
  Object {
    "type": "SET_USER_DATA",
    "userData": Object {
      "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
      "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
      "bio": "populated",
      "dob": 1615046987793,
      "gender": "gender",
      "name": null,
    },
    "userId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
  },
]
`;

exports[`async actions create SET_AVATAR_DATA when fetching has been done 1`] = `
Array [
  Object {
    "avatar": Object {
      "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
    },
    "type": "SET_USER_AVATAR",
    "userId": "22bb3fe5-185c-4c87-8633-1586586506b7",
  },
]
`;

exports[`async actions create SET_USER_DATA when fetching has been done 1`] = `
Array [
  Object {
    "type": "SET_USER_DATA",
    "userData": Object {
      "userData": Object {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517",
        "bio": "populated",
        "dob": 1615046987793,
        "gender": "gender",
        "name": null,
      },
    },
    "userId": "22bb3fe5-185c-4c87-8633-1586586506b7",
  },
]
`;
